### PR TITLE
Add Chatbot Pattern to Site

### PR DESCRIPTION
**Pull request by vincentsnagg**
Hi there 👋🏽! This PR adds Chatbot Pattern 🤖 💬 guidance into Carbon, using design and content guidelines from Watson Health, Watson, and the Watson Assistant team. I've been chatting with Tyler and Connor about using this PR as a place to get thoughts from the broader team and fill in any gaps. 

To close an issue, I think I would need to move the initial issue that is currently  in the Carbon Contributing Repo into this Website repo...

#### Changelog

**New**

- Adds Chatbot Pattern Guidance to the site
   - Adds in the 3 Tabs (Overview, Usage, and Content) based on Tyler's Pattern document

Closes: #1745

### CODE REVIEW

**Comment by designertyler** [comment]
@vincentsnagg I added an issue here with some of my feedback. I'll get with you to see where I can help. Thanks again for getting this PR submitted! https://github.com/carbon-design-system/carbon-website/issues/1745

**Comment by jnm2377** [comment]
Hi @vincentsnagg thanks for taking this on! Just wondering if you've addressed all of @designertyler feedback so this can be re-reviewed, or if you're still working on this. @connor-leech can you also get eyes on this?

Reviewers: In the future, can we just keep the feedback in the same PR rather than opening an issue, thanks! :)

File: src/content/patterns/chatbot/content.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+title: Chatbots
+tabs: ['Overview', 'Usage', 'Content']
+---
+
+### You have established a need for a chatbot based on extensive market validation and user research. You have also determined the scope of your bot and the topics it will be able to discuss with users. How should you begin to think about designing your conversation?
```
Comment (connor-leech) [inline src/content/patterns/chatbot/content.mdx]:
```suggestion
### You have established a need for a chatbot based on extensive market validation and user research. How should you begin to think about designing your conversation?
```

This was a bit long as a header. I think we can drop the middle sentence and still have the same impact.

**Comment by aagonzales** [comment]
Two things:
- move this pattern under the experimental section
- I think a lot of the images would benefit from using the 12 column image component instead of 8. In the markdown code you can just change the col # to 12, its super easy.
   - for 8 column image component use width: 736px
   - for 12 column image component use width: 1120px
   - Also make sure the images are exported at 2x

**Comment by vincentsnagg** [comment]
@jnm2377 I've got one more piece of @designertyler's feedback to incorporate as well as @aagonzales' feedback to incorporate, and then this should be ready for one more review. Thanks all!

**Comment by vincentsnagg** [comment]
@jnm2377 I've aligned with @designertyler and @connor-leech, and incorporated @aagonzales' feedback. This should be good to go for another review!

small thing @connor-leech  — does "Chatbot" or "Chatbots" sound better in the navigation (I proposed Chatbots in the last commit because it aligns with the titles on the pattern pages, but of course feel free to make the changes that work the best!)

**Comment by designertyler** [comment]
Great work @vincentsnagg ! Thanks so much for getting this together 🎉

**Pull request merged**